nixos/modules/services/continuous-integration/github-runner/service.nix

```
@@ -170,6 +170,9 @@ with lib;
     # If running in ephemeral mode, restart the service on-exit (i.e., successful de-registration of the runner)
     # to trigger a fresh registration.
     Restart = if cfg.ephemeral then "on-success" else "no";
+    # If the runner exits with `ReturnCode.RetryableError = 2`, always restart the service:
+    # https://github.com/actions/runner/blob/40ed7f8/src/Runner.Common/Constants.cs#L146
+    RestartForceExitStatus = "2";
```

I know this would get converted to strings, but I'm not sure what's more "proper" here. Thoughts?

```suggestion
    RestartForceExitStatus = 2;
```

---

I think the proper thing would be `List[int]` type, ref. the [documentation](https://www.freedesktop.org/software/systemd/man/systemd.service.html#RestartForceExitStatus=) which says `Takes a list of exit status definitions [...]`. But I'm personally fine with using a string here, similar to the existing usages in nixpkgs.

---

Strictly speaking its `List[int | str]`.

Changed it to a `[ 2 ]`.